lib/types/subnet_info.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Types.SubnetInfo do
+  @moduledoc """
+  Struct to hold subnet attestations for easier db storing:
+  - data: An attestation data.
+  - attestations: List of all the collected Attestations.
+  """
+
+  defstruct [:data, :attestations]
+
+  @type t :: %__MODULE__{
+          data: Types.AttestationData.t(),
+          attestations: list(Types.Attestation.t())
+        }
+
+  @spec new_subnet_with_attestation(Types.Attestation.t()) :: Types.SubnetInfo.t()
+  def new_subnet_with_attestation(%Types.Attestation{data: data} = attestation),
+    do: %__MODULE__{data: data, attestations: [attestation]}
+
+  @spec aggregate_attestation(t(), Types.Attestation.t()) :: t()
```

Comment:
This doesn't aggregate (e.g. doesn't aggregate signatures). Let's call it something like `add_attestation`. Let's also document the function to explain what the function does (adds new attestations to the struct, but only if their data matches the base one).